### ISSUE TITLE
[google-beta] google_container_node_pool should accept a netmask on network_config.pod_ipv4_cidr_block

### ISSUE THREAD

**Issue by mwebler** [hashicorp/terraform-provider-google#11024]
Hi! On the google-beta provider there is an issue with the validation of the `google_container_node_pool` resource. In the `network_config` block, `pod_ipv4_cidr_block` should accept a /netmask only to automatically create a range of the configured size but the validation is expecting a CIDR range only. 

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version
```
Terraform v1.1.3
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.6.0
+ provider registry.terraform.io/hashicorp/google-beta v4.6.0
```

### Affected Resource(s)

- resource google_container_node_pool

### Terraform Configuration Files

```tf
resource "google_container_cluster" "primary" {
  name     = "my-gke-cluster"
  location = "us-central1"

  remove_default_node_pool = true
  initial_node_count       = 1
}

resource "google_container_node_pool" "primary_preemptible_nodes" {
  name       = "my-node-pool"
  cluster    = google_container_cluster.primary.id
  node_count = 1

  node_config {
    preemptible  = true
    machine_type = "e2-medium"
  }

  network_config {  
    create_pod_range = true
    pod_ipv4_cidr_block = "/16"
    pod_range = "my_pods_range"
  }
}

```

### Output

`network_config.0.pod_ipv4_cidr_block" is not a valid IP CIDR range: invalid CIDR address: /16`

### Expected Behavior

To accept a netmask like /16 to create a range of that size, as stated in the docs. Both the [Docs](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#pod_ipv4_cidr_block) and [variable description](https://github.com/hashicorp/terraform-provider-google-beta/blob/faf7b04808eed7cdde77612056a7dd5b022e763f/google-beta/resource_container_node_pool.go#L250) say a /netmask should be allowed.

### Actual Behavior
`pod_ipv4_cidr_block` is expecting a CIDR range. [Master branch validation code](https://github.com/hashicorp/terraform-provider-google-beta/blob/faf7b04808eed7cdde77612056a7dd5b022e763f/google-beta/resource_container_node_pool.go#L249).

### Steps to Reproduce

Create a GKE cluster and node pool, setting the node pool's `pod_ipv4_cidr_block` to any /netmask (like /14).

1. `terraform plan`

### Additional info
The `google_container_cluster` resource has a [similar option and accepts /netmask values correctly - no CIDR validation](https://github.com/hashicorp/terraform-provider-google-beta/blob/faf7b04808eed7cdde77612056a7dd5b022e763f/google-beta/resource_container_cluster.go#L950-L958)

**Comment by slevenick** [comment]
Huh... So I tested this out and this field has some weird behavior if you only add a /netmask. For example:

I created a node pool with the following network_config:
```
  "networkConfig": {
   "createPodRange": true,
   "podIpv4CidrBlock": "/16",
   "podRange": "auto-pod-range"
  }
```

And the API returned: 
```
 "networkConfig": {
  "podRange": "auto-pod-range",
  "podIpv4CidrBlock": "10.32.0.0/16"
 },
```

Which causes TF to detect a diff and attempt to recreate the cluster. So at first glance it doesn't seem like the API supports netmasks alone. Are you aware of a situation where specifying only the netmask is required or desirable?

**Comment by mwebler** [comment]
In this case, allowing the node pool to create its own range is just a way to avoid dealing with configuring the ranges myself and letting GCP handle the work of finding an available range (as other clusters in the network already created pods and services ranges automatically).

For now I have been leaving the `pod_ipv4_cidr_block` blank, so Google creates a new range with the default size (/14). This seems to work fine and I didn't see any problems with TF so far.

e.g. using the following network_config on Terraform:
```
network_config {  
    create_pod_range = true
    pod_range = "my_pods_range"
  }
```

My .tfstate file looks like this after the cluster is created and no attempt to recreate the cluster happens on future `terraform plan` or `terraform apply` runs:
```
"network_config": [
  {
    "create_pod_range": true,
    "pod_ipv4_cidr_block": "10.152.0.0/14",
    "pod_range": "my_pods_range"
  }
],
```

**Comment by slevenick** [comment]
Yeah, there's a difference between leaving `pod_ipv4_cidr_block` blank because this field is marked as `Computed` which allows it to take a default from the API. If you specify `pod_ipv4_cidr_block = "/14"` and the API returns `pod_ipv4_cidr_block = "10.152.0.0/14"` Terraform will complain they don't match and attempt to recreate it. This could be patched with a diff suppress func that compares the mask if the user specified only a mask. Does that sound like it will solve your problem?

**Comment by mwebler** [comment]
yes, that would probably be enough given that works fine [for the cluster ranges ](https://github.com/hashicorp/terraform-provider-google-beta/blob/faf7b04808eed7cdde77612056a7dd5b022e763f/google-beta/resource_container_cluster.go#L950-L958)using the `cidrOrSizeDiffSuppress` function.

**Comment by kevinsqi** [comment]
Would be great to get a fix for this, I'm seeing it as well.

**Comment by IanMoroney** [comment]
I am also having this issue, so it would be great to have a solution

**Comment by jbbqqf** [comment]
Hi! Triaging older GKE network_config requests — this one looks shipped.

On `google_container_node_pool.network_config.pod_ipv4_cidr_block`, the strict CIDR validator has been removed and replaced with a `CidrOrSizeDiffSuppress` plus a description that explicitly allows the `/netmask` form. See [`google-beta/services/container/resource_container_node_pool.go:576`](https://github.com/hashicorp/terraform-provider-google-beta/blob/main/google-beta/services/container/resource_container_node_pool.go#L576):

> Set to blank to have a range chosen with the default size. Set to /netmask (e.g. /14) to have a range chosen with a specific netmask. Set to a CIDR notation (e.g. 10.96.0.0/14) to pick a specific range to use.

Working snippet:

```hcl
resource "google_container_node_pool" "primary" {
  provider = google-beta
  name     = "my-node-pool"
  cluster  = google_container_cluster.primary.id

  network_config {
    create_pod_range    = true
    pod_ipv4_cidr_block = "/16"   # accepted, range will be auto-allocated at /16
    pod_range           = "my_pods_range"
  }
}
```

Provider docs: https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/resources/container_node_pool#nested_network_config

Would you mind closing this out? If you still hit a validation error on a current provider version with `/14` or `/16`, drop the exact error and I'll dig.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the schema was verified manually against the current `terraform-provider-google-beta` `main` branch.)
